components/primitives/Form/Input.tsx

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import { cn } from "@/lib/utils/cn";
+
+interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {}
+
+const Input = React.forwardRef<HTMLInputElement, InputProps>(
+  ({ id, type, className, ...rest }, ref) => {
+    const [showPassword, setShowPassword] = React.useState(false);
+    return (
+      <div className="relative">
+        <input
+          className={cn(
+            "py-2 px-3 text-sm placeholder:text-sm block w-full border border-[#D0D5DD] bg-white rounded-md outline-0 focus:border-abeg-green-20 placeholder:text-abeg-neutral-50 text-abeg-neutral-10 disabled:bg-[#F0F2F5] disabled:border-[#D0D5DD]",
+            {
+              "pr-[32px]": type === "password",
+            },
+            className,
+          )}
+          type={type === "password" && showPassword ? "text" : type}
+          ref={ref}
+          id={id}
+          {...rest}
+        />
+        {type === "password" && (
+          <svg
+            xmlns="http://www.w3.org/2000/svg"
+            fill="none"
+            viewBox="0 0 24 24"
+            strokeWidth={1.5}
+            stroke="currentColor"
+            className="w-5 h-5 text-abeg-neutral-50 absolute left-[calc(100%-20px)] top-1/2 -translate-x-1/2 -translate-y-1/2"
+            onClick={() => setShowPassword((curPassword) => !curPassword)}
```

Comment:
```suggestion
            onClick={() => ref.current.type = ref.current.type === "password" ? "text" : "password";}
```

...and you can have the `ref` change the input `type` onclick without using the `state`.